docs/api/ethereum.md

```diff
@@ -1030,6 +1045,256 @@ A class instance `variable` holding a `string` which represents smart contract a
 
 A class instance `variable` holding a `string` which represents the URL to the compiled ERC-20 related smart contract definition file. This file is used when deploying new value ledgers to the network.
 
+## Deploy gateway
+
+Deploy gateway allows for delegating deployment of an `AssetLedger` smart contract to a third party and transfering value in the same atomic transaction.
+
+To perform a delegate deploy through `DeployGateway`, you need to follow its flow:
+
+1. Maker (address creating the delegate deploy) defines the deploy (defines the data for the `AssetLedger` that will be created, who will be the owner of that `AssetLedger` and what amount of tokens will be sent to what address in exchange).
+2. Maker generates the deploy claim and signs it (`claim` functions).
+3. Maker approves value transfer.
+4. Maker sends deploy and signature to the Taker trough arbitrary channel.
+6. Taker performs the deploy.
+
+`Deploy` class is responsible for defining what will happen in the atomic swap. We support two different deploy configurations which we will call a fixed deploy and a dynamic deploy.
+
+In fixed deploy, the Taker of the deploy (its wallet address) is known, and we want to make an atomic deploy specifically with him and only him. For this, we need to set `deploy.takerId`.
+
+In dynamic deploy, we do not care who performs the deploy. In this case, we do not set `deploy.takerId`. Now any account (wallet) will be able to perform such deploy and will automatically become its Taker.
+
+### DeployGateway(provider, deployGatewayId)
+
+A `class` which represents a smart contract on the Ethereum blockchain.
+
+**Arguments**
+
+| Argument | Description
+|-|-
+| deployGatewayId | [required] A `string` representing an address of the [0xcert deploy gateway smart contract](#public-addresses) on the Ethereum blockchain.
+| provider | [required] An instance of an provider.
+
+**Usage**
+
+```ts
+import { MetamaskProvider } from '@0xcert/ethereum-metamask-provider';
+import { DeployGateway } from '@0xcert/ethereum-deploy-gateway';
+
+// arbitrary data
+const provider = new MetamaskProvider();
+const deployGatewayId = '0x073d230a53bffc8295d9a5247296213298e3fbcf';
+
+// create deploy gateway instance
+const deployGateway = new DeployGateway(provider, deployGatewayId);
+```
+
+### cancel(deploy)
+
+An `asynchronous` class instance `function` which marks the provided `deploy` as canceled. This prevents the `deploy` to be performed.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| deploy.assetLedgerData.capabilities | [required] An array of `AssetLedgerCapability` representing the capabilities of the `AssetLedger` that will be deployed.
+| deploy.assetLedgerData.name | [required] A `string` representing the name of the `AssetLedger` that will be deployed.
+| deploy.assetLedgerData.owner | [required] A `string` representing the address of owner of the `AssetLedger` that will be deployed.
```

Comment:
| deploy.assetLedgerData.owner | [required] A `string` representing the address of the owner of the `AssetLedger` that will be deployed.